paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerAdapter.kt

```diff
@@ -46,7 +46,7 @@ interface CustomerAdapter {
      * @param paymentOption, the [PaymentOption] to save to the data store. If null, the selected
      * payment method option will be cleared from the data store.
      */
-    suspend fun setSelectedPaymentOption(paymentOption: PaymentOption?)
+    suspend fun setSelectedPaymentOption(paymentOption: PaymentOption?): Result<Boolean>
```

Comment:
Should this just be 
```suggestion
    suspend fun setSelectedPaymentOption(paymentOption: PaymentOption?): Boolean
```

If not, when would you return `Result.success(false)`?

Reply:
It does't feel right when you look at it this way. I guess the only way you should get this is if this invocation was unable to set the selected payment option. `Result.failure` could be reserved for exceptions?

Reply:
Maybe `Result<Unit>` then?

Reply:
Maybe we could return Result<PaymentOption> to indicate that it was successful or not